src/ethereum/ethNetwork.js

```diff
@@ -1447,451 +1265,6 @@ export class EthereumNetwork {
-        }
-        const amberdataTxs = cleanedResponseObj.payload.records
-        for (const amberdataTx of amberdataTxs) {
-          try {
-            const cleanAmberdataTx = asAmberdataAccountsTx(amberdataTx)
-
-            const tx = this.processAmberdataTxRegular(
-              cleanAmberdataTx,
-              currencyCode
-            )
-            if (tx) {
-              allTransactions.push(tx)
-            }
-          } catch (e) {
-            this.ethEngine.error(
-              `checkTxsAmberdata process regular ${safeErrorMessage(
-                e
-              )}\n${JSON.stringify(amberdataTx)}`
-            )
-            throw new Error('checkTxsAmberdata regular amberdataTx is invalid')
-          }
-        }
-        if (amberdataTxs.length === 0) {
-          break
-        }
-        page++
-      } else {
-        let cleanedResponseObj: FetchGetAmberdataApiResponse
-        try {
-          if (startDate) {
-            url = url + `&startDate=${startDate}&endDate=${Date.now()}`
-          }
-          const jsonObj = await this.fetchGetAmberdataApi(url)
-          cleanedResponseObj = asFetchGetAmberdataApiResponse(jsonObj)
-        } catch (e) {
-          this.logError('checkTxsAmberdata internal txs', e)
-          throw new Error('checkTxsAmberdata (internal tx) response is invalid')
-        }
-        const amberdataTxs = cleanedResponseObj.payload.records
-        for (const amberdataTx of amberdataTxs) {
-          try {
-            const cleanamberdataTx = asAmberdataAccountsFuncs(amberdataTx)
-            const tx = this.processAmberdataTxInternal(
-              cleanamberdataTx,
-              currencyCode
-            )
-            if (tx) {
-              allTransactions.push(tx)
-            }
-          } catch (e) {
-            this.ethEngine.error(
-              `checkTxsAmberdata process internal ${safeErrorMessage(
-                e
-              )}\n${JSON.stringify(amberdataTx)}`
-            )
-            throw new Error('checkTxsAmberdata internal amberdataTx is invalid')
-          }
-        }
-        if (amberdataTxs.length === 0) {
-          break
-        }
-        page++
-      }
-    }
-
-    return allTransactions
-  }
-
-  async checkTxsAmberdata(
-    params: GetTxsParams
-  ): Promise<EthereumNetworkUpdate> {
-    const { startBlock, startDate, currencyCode } = params
-    const allTxsRegular: EdgeTransaction[] = await this.getAllTxsAmberdata(
-      startBlock,
-      startDate,
-      currencyCode,
-      true
-    )
-
-    const allTxsInternal: EdgeTransaction[] = await this.getAllTxsAmberdata(
-      startBlock,
-      startDate,
-      currencyCode,
-      false
-    )
-
-    return {
-      tokenTxs: {
-        [`${this.currencyInfo.currencyCode}`]: {
-          blockHeight: startBlock,
-          edgeTransactions: [...allTxsRegular, ...allTxsInternal]
-        }
-      },
-      server: 'amberdata'
-    }
-  }
-
-  async checkTxsBlockbook(
```

Comment:
This function is called checkTxs but it does more than that because the single api call can fill an entire ethereum network update. You can keep it still use it to get the nonce and balances but remove the tokenTxs if they aren't accurate for internal txs.